cmd/jaeger/internal/integration/badger_test.go

```diff
@@ -23,7 +23,7 @@ func TestBadgerStorage(t *testing.T) {
 			GetOperationsMissingSpanKind: true,
 		},
 	}
-	s.e2eInitialize(t, "badger")
+	s.e2eInitialize(t, "badger", true)
```

Comment:
let's do this differently
  * name the parameter such that its default value (false) works for almost every test, such that you only override it for Kafka
  * define that parameter in the struct, not in the function. You can see above `GetOperationsMissingSpanKind` is an example of parameter that affects how the tests execute.

Reply:
okay will work on this

Reply:
Introduced a new parameter `SkipStorageCleaner` inside the `E2EStorageIntegration` which is by default false. So all the other tests work fine. We'll be overriding it only for the Kafka collector config so we that we dont inject the storage cleaner extension.  I'd done this in my latest fix, let me know if this is fine